packages/hw-transport-node-hid-noevents/src/TransportNodeHid.ts

```diff
@@ -45,14 +42,12 @@ export default class TransportNodeHidNoEvents extends Transport<?string> {
   /**
    *
    */
-  static list = (): Promise<(?string)[]> =>
+  static list = (): Promise<any> =>
     Promise.resolve(getDevices().map((d) => d.path));
 
   /**
    */
-  static listen = (
-    observer: Observer<DescriptorEvent<?string>>
-  ): Subscription => {
+  static listen = (observer: Observer<DescriptorEvent<any>>): Subscription => {
```

Comment:
Is `Subscription` an `rxjs` Subsciption? hard to tell since it is imported from `hw-transport` and not rxjs.

Anyway, I think this method is kind of awkward, I think it better if the method returned an `Observable<Event>` that you subscribe to yourself, instead of accepting an observer as arg and returning a subscription. The reason for this is that if we were to return an Observable we can pipe it! And do stuff with it, whereas a Subscription is.... well end of the road really.

Also the `DescriptorEvent<any>` should probably be massaged a bit. Better if it is a tagged union without the `any` generic?

Maybe:
```typescript
type LedgerAvailabilityEvent = LedgerAvailabilityEventAvailable | LedgerAvailabilityEventUnavailable

type LedgerAvailabilityEventAvailable = Readonly<{
    type: 'new'
    // Device that was recently found
   device: Device
}>

type LedgerAvailabilityEventUnavailable = Readonly<{
    type: 'lost'
    // Device that is not available anymore
   device: Device
}>
```

Or similar.

Reply:
I agree with your feedback, we should take note for future but right now we are porting code without changing the interface.
To me it's still discussable: It was initially meant to be loosely coupled with rxjs: someone that depends on a hw-transport-* do not bring the rxjs package in with the whole bundle. That way, you can even still use this without rxjs.

agreed for union type.